File: binary_search.py
# binary_search.py

def binary_search(arr, x):
    low = 0
    high = len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] < x:
            low = mid + 1
        elif arr[mid] > x:
            high = mid - 1
        else:
            return mid
    return -1

# Örnek kullanım
if __name__ == "__main__":
    arr = [2, 3, 4, 10, 40]
    x = 10
    result = binary_search(arr, x)
    if result != -1:
        print("Element is present at index", result)
    else:
        print("Element is not present in array")

File: cppAlgorithms/BinarySearch.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int binarySearch(const std::vector<int>& arr, int target) {
    int left = 0, right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target)
            return mid;
        if (arr[mid] < target)
            left = mid + 1;
        else
            right = mid - 1;
    }
    return -1; //element not found
}

// Dizi Yazdırma Fonksiyonu
void printArray(const vector<int>& arr) {
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    vector<int> arr = {3932, 932474, 562, 5, 123456789};

    cout << "Siralanmis Dizi: ";
    printArray(arr); // Diziyi yazdır

    int target = 10;
    sort(arr.begin(), arr.end()); // Ensure the array is sorted
    int result = binarySearch(arr, target);

    if (result != -1)

        cout << "Element found at index " << result << endl;
    else
        cout << "Element not found in the array" <<endl;

    return 0;
}
